agent/rpc/peering/service.go

```diff
@@ -370,9 +370,27 @@ func (s *Service) PeeringList(ctx context.Context, req *pbpeering.PeeringListReq
 	if err != nil {
 		return nil, err
 	}
+
+	for _, p := range peerings {
+		p.State = s.reconciledStreamStateHint(p.ID, p.State)
+	}
 	return &pbpeering.PeeringListResponse{Peerings: peerings}, nil
 }
 
+// TODO(peering): Maybe get rid of this when actually monitoring the stream health
+// reconciledStreamStateHint peaks into the streamTracker and determines whether a peering should be marked
+// as PeeringState.Active or not
+func (s *Service) reconciledStreamStateHint(pID string, pState pbpeering.PeeringState) pbpeering.PeeringState {
+	streamState, found := s.streams.streamStatus(pID)
+
+	if found && streamState.Connected {
+		return pbpeering.PeeringState_ACTIVE
```

Comment:
Maybe this is not enough to just say that the streamState is connected.

Maybe we need to say something like "if no errors in the last 10 seconds received or sent" then it is active and not failing." In that case, the corollary would give the failing state hint : "if one or more errors were received or sent, then it is active but failing. " and we'd mark the state as PeeringState_Failing.

Reply:
For the time being I think this is fine. Once we have the stream heartbeating then we can revisit the connected state implying Active with some better backing data.